x/orderbook/bucket_test.go

```diff
@@ -1,138 +1,289 @@
+		Taker:       weavetest.NewCondition().Address(),
+		Maker:       weavetest.NewCondition().Address(),
+		TakerPaid:   coin.NewCoinp(100, 0, "ETH"),
+		MakerPaid:   coin.NewCoinp(7, 234456, "BTC"),
+		ExecutedAt:  now,
+	}
+
 	cases := map[string]struct {
-		obj orm.Object
-		wantErr *errors.Error 
+		obj      orm.Object
+		expected []byte
+		wantErr  *errors.Error
 	}{
 		"success": {
-			obj: orm.NewSimpleObj(nil, new(Trade)), 
-			wantErr: nil,
+			obj:      orm.NewSimpleObj(nil, trade),
+			expected: trade.OrderID,
+			wantErr:  nil,
 		},
+		"failure, obj is nil": {
+			obj:      nil,
+			expected: nil,
+			wantErr:  nil,
+		},
+		// TODO add obj.Value() nil case
 		"not trade": {
-			obj: orm.NewSimpleObj(nil, new(OrderBook)),
-			wantErr: errors.ErrState,
+			obj:      orm.NewSimpleObj(nil, new(OrderBook)),
+			expected: nil,
+			wantErr:  errors.ErrState,
 		},
 	}
 
 	for testName, tc := range cases {
 		t.Run(testName, func(t *testing.T) {
-			if _, err := orderIDIndexer(tc.obj); !tc.wantErr.Is(err) {
+			index, err := orderIDIndexer(tc.obj)
+
+			if !tc.wantErr.Is(err) {
 				t.Fatalf("unexpected error: %+v", err)
 			}
+			assert.Equal(t, tc.expected, index)
 		})
 	}
 }
 
 func TestOrderBookTimedIndexer(t *testing.T) {
 	now := weave.AsUnixTime(time.Now())
-	invalidTime:= weave.AsUnixTime(time.Unix(-1, 0))
+	invalidTime := weave.AsUnixTime(time.Unix(-1, 0))
 
 	validTrade := &Trade{
-				Metadata:    &weave.Metadata{Schema: 1},
-				OrderID:     weavetest.SequenceID(14),
-				OrderBookID: weavetest.SequenceID(2),
-				Taker:       weavetest.NewCondition().Address(),
-				Maker:       weavetest.NewCondition().Address(),
-				TakerPaid:   coin.NewCoinp(100, 0, "ETH"),
-				MakerPaid:   coin.NewCoinp(7, 234456, "BTC"),
-				ExecutedAt:  now,
-			}
-				
+		Metadata:    &weave.Metadata{Schema: 1},
+		OrderID:     weavetest.SequenceID(14),
+		OrderBookID: weavetest.SequenceID(2),
+		Taker:       weavetest.NewCondition().Address(),
+		Maker:       weavetest.NewCondition().Address(),
+		TakerPaid:   coin.NewCoinp(100, 0, "ETH"),
+		MakerPaid:   coin.NewCoinp(7, 234456, "BTC"),
+		ExecutedAt:  now,
+	}
+
 	invalidTrade := &Trade{
-				Metadata:    &weave.Metadata{Schema: 1},
-				OrderID:     weavetest.SequenceID(14),
-				OrderBookID: weavetest.SequenceID(2),
-				Taker:       weavetest.NewCondition().Address(),
-				Maker:       weavetest.NewCondition().Address(),
-				TakerPaid:   coin.NewCoinp(100, 0, "ETH"),
-				MakerPaid:   coin.NewCoinp(7, 234456, "BTC"),
-				ExecutedAt:  invalidTime,
-			}
+		Metadata:    &weave.Metadata{Schema: 1},
+		OrderID:     weavetest.SequenceID(13),
+		OrderBookID: weavetest.SequenceID(2),
+		Taker:       weavetest.NewCondition().Address(),
+		Maker:       weavetest.NewCondition().Address(),
+		TakerPaid:   coin.NewCoinp(100, 0, "ETH"),
+		MakerPaid:   coin.NewCoinp(7, 234456, "BTC"),
+		ExecutedAt:  invalidTime,
+	}
+
+	successCaseExpectedValue := BuildOrderBookTimedIndexTestCase(validTrade)
```

Comment:
Again, I would hard-code this.
It only works if the time is also hardcoded...

Eg. 

```golang
executed := weave.UnixTime(1555777200)  // this is April 2019
// note we use orderbook id, which is a bug in original code... so doing the manual checks does help us :)
// hex(1555777200) = 0x5cbb46b0
successCaseExpectedValue := []byte{0, 0, 0, 0, 0, 0, 0, 2, 0, 0, 0, 0, 0x5c, 0xbb, 0x46, 0xb0}
```